apps/server/src/modules/news/repo/news.repo.ts

```diff
@@ -1,45 +1,55 @@
 import { Injectable } from '@nestjs/common';
-import { News } from '../entity/news.entity';
-import { PaginationModel } from '../../../shared/core/repo/interface/pagination.interface';
 import { EntityManager } from '@mikro-orm/mongodb';
-import { EntityId } from '../../../shared/domain';
-import { ICreateNewsDto } from '../uc';
-import { Dictionary } from '@mikro-orm/core';
+import { EntityId, IPagination } from '@shared/domain';
+import { Counted } from '@shared/domain/types';
+import { News } from '../entity';
+import { NewsScope } from './news-scope';
+import { NewsTargetFilter } from './news-target-filter';
 
 @Injectable()
 export class NewsRepo {
 	constructor(private readonly em: EntityManager) {}
 
-	async create(props: ICreateNewsDto): Promise<News> {
-		const news = this.em.create(News, props);
-		this.em.persistAndFlush(news);
+	async save(news: News): Promise<News> {
+		await this.em.persistAndFlush(news);
 		return news;
 	}
 
-	async findAllByUser(userId: EntityId, pagination: PaginationModel = {}): Promise<News[]> {
-		// TODO filter by user scopes
-		const newsList = this.em.find(
-			News,
-			{},
-			{
-				populate: ['school', 'creator', 'updater'],
-				...pagination,
-			}
-		);
-		return newsList;
+	/**
+	 * Find news
+	 * @param targets
+	 * @param unpublished
+	 * @param pagination
+	 */
+	async findAll(
+		targets: NewsTargetFilter[],
+		unpublished: boolean,
+		pagination: IPagination = {}
+	): Promise<Counted<News[]>> {
+		const scope = new NewsScope();
+		scope.byTargets(targets);
+		scope.byUnpublished(unpublished);
+		// TODO sort by publishedAt?
+		// scope.sortBy(...)
+		const countedNewsList = await this.findNewsAndCount(scope.query, pagination);
+		return countedNewsList;
 	}
 
 	/** resolves a news document with some elements names (school, updator/creator) populated already */
 	async findOneById(id: EntityId): Promise<News> {
-		const news = this.em.findOneOrFail(News, id, ['school', 'creator', 'updater']);
+		const news = await this.em.findOneOrFail(News, id, ['school', 'creator', 'updater']);
 		return news;
 	}
 
-	update(id: EntityId, props: Dictionary) {
-		return `This action updates a #${id} news`;
+	async delete(news: News): Promise<void> {
+		await this.em.removeAndFlush(news);
 	}
 
-	remove(id: EntityId) {
-		return `This action removes a #${id} news`;
+	private async findNewsAndCount(query, pagination: IPagination): Promise<Counted<News[]>> {
+		const [obj, count] = await this.em.findAndCount(News, query, {
+			...pagination,
```

Comment:
For internal calls to uc the pagination is optional, at the moment it look like it is not possible to call the uc without pagination and get not limited data.

Reply:
currently no case without it exists. happily this is simple to change? when we require it.